heron/executor/src/python/heron_executor.py

```diff
@@ -147,6 +148,7 @@ def __init__(self, args, shell_env):
         base64.b64decode(parsed_args.instance_jvm_opts.lstrip('"').
                          rstrip('"').replace('&equals;', '='))
     self.classpath = parsed_args.classpath
+    self.master_host = os.environ.get('HOST') if 'HOST' in os.environ else socket.gethostname()
```

Comment:
As far as I can find, you can't natively "introspect" the host IP from within a Docker container. The `HOST` environment variable is set by Marathon automatically as an environment variable for use cases like this where we care about the HOST IP. So, for other environments outside of DC/OS and Marathon, it will be a requirement (if the executor is running inside a Docker container) to set the HOST environment variable when launched, otherwise the hostname will default to the container ID (which was the trouble I ran into originally).